os/include/tinyara/lcd/lcd.h

```diff
@@ -88,39 +102,95 @@ struct lcd_planeinfo_s {
 	/* LCD Data Transfer ******************************************************/
 	/* This method can be used to write a partial raster line to the LCD:
 	 *
+	 *  dev     - LCD interface to write to
 	 *  row     - Starting row to write to (range: 0 <= row < yres)
 	 *  col     - Starting column to write to (range: 0 <= col <= xres-npixels)
 	 *  buffer  - The buffer containing the run to be written to the LCD
 	 *  npixels - The number of pixels to write to the LCD
 	 *            (range: 0 < npixels <= xres-col)
 	 */
 
-	int (*putrun)(fb_coord_t row, fb_coord_t col, FAR const uint8_t *buffer,
-				  size_t npixels);
+	int (*putrun)(FAR struct lcd_dev_s *dev, fb_coord_t row, fb_coord_t col, FAR const uint8_t *buffer, size_t npixels);
+
+	/* This method can be used to write a rectangular area to the LCD:
+	 *
+	 *  dev       - LCD interface to write to
+	 *  row_start - Starting row to write to (range: 0 <= row < yres)
+	 *  row_end   - Ending row to write to (range: row_start <= row < yres)
+	 *  col_start - Starting column to write to (range: 0 <= col <= xres)
+	 *  col_end   - Ending column to write to
+	 *              (range: col_start <= col_end < xres)
+	 *  buffer    - The buffer containing the area to be written to the LCD
+	 *  stride    - Length of a line in bytes. This parameter may be necessary
+	 *              to allow the LCD driver to calculate the offset for partial
+	 *              writes when the buffer needs to be splited for row-by-row
+	 *              writing.
+	 *
+	 * NOTE: this operation may not be supported by the device, in which case
+	 * the callback pointer will be NULL. In that case, putrun() should be
+	 * used.
+	 */
+
+	int (*putarea)(FAR struct lcd_dev_s *dev, fb_coord_t row_start, fb_coord_t row_end, fb_coord_t col_start, fb_coord_t col_end, FAR const uint8_t *buffer, fb_coord_t stride);
```

Comment:
I think putarea is almost same as putrun. any special reason why you add this?

Reply:
putarea is for drawing a rectangle (x1,x2,y1,y2)
putrun just draws a line at a point(x,y) 
If user calls putrun multiple times instead of putarea, he needs to do some calculations ? If it is so, I will remove it